bugsnag-android-core/src/main/java/com/bugsnag/android/SessionFilenameInfo.kt

```diff
@@ -27,29 +29,52 @@ internal data class SessionFilenameInfo(
          * Generates a filename for the session in the format
          * "[UUID][timestamp]_v2.json"
          */
-        fun toFilename(timestamp: Long, uuid: String): String {
-            return "${uuid}${timestamp}_v2.json"
+        fun toFilename(apiKey: String, timestamp: Long, uuid: String): String {
+            return "${apiKey}_${uuid}${timestamp}_v2.json"
         }
 
         @JvmStatic
-        fun defaultFilename(): String {
-            return toFilename(System.currentTimeMillis(), UUID.randomUUID().toString())
+        fun defaultFilename(apiKey: String, config: ImmutableConfig): String {
+            val sanitizedApiKey = apiKey.takeUnless { it.isEmpty() } ?: config.apiKey
+
+            return toFilename(
+                sanitizedApiKey,
+                System.currentTimeMillis(),
+                UUID.randomUUID().toString()
+            )
         }
 
-        fun fromFile(file: File): SessionFilenameInfo {
+        fun fromFile(file: File, config: ImmutableConfig): SessionFilenameInfo {
             return SessionFilenameInfo(
+                findApiKeyInFilename(file, config),
                 findTimestampInFilename(file),
                 findUuidInFilename(file)
             )
         }
 
         private fun findUuidInFilename(file: File): String {
-            return file.name.substring(0, uuidLength - 1)
+            val uuidWithTimestamp = file.name.substringAfter("_")
+            val uuidName = if (uuidWithTimestamp.length >= uuidLength) {
+                uuidWithTimestamp.take(uuidLength)
+            } else {
+                null
+            }
+            return uuidName.takeUnless { it.isNullOrBlank() } ?: ""
         }
 
         @JvmStatic
         fun findTimestampInFilename(file: File): Long {
-            return file.name.substring(uuidLength, file.name.indexOf("_")).toLongOrNull() ?: -1
+            return file.name.split("_")[1].drop(uuidLength)
+                .toLongOrNull() ?: -1
```

Comment:
it will be 
```
 return file.name.substringAfter('_')
                .drop(uuidLength)
                .substringBefore('_')
                .toLongOrNull() ?: -1
```
in order to move the last part of file name after v2